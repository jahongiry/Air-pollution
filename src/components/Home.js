import { useSelector } from 'react-redux';
import React, { useEffect, useState } from 'react';
import classes from './Home.module.css';
import CharactersHome from './CharactersHome';

function Home() {
  const result = useSelector((state) => state.card);
  const [query, setQuery] = useState('');
  const [visible, setVisible] = useState(result);

  const inputRef = React.createRef();
  const updateQuery = () => {
    const inputText = inputRef.current.value;
    setQuery(inputText);
  };

  useEffect(() => {
    setVisible(result);
  }, [result]);

  useEffect(() => {
    const searchedName = query.toLowerCase();
    // eslint-disable-next-line
    setVisible(result.filter((info) => {
      const apiNames = info.names.toLowerCase();
      if (apiNames.includes(searchedName)) {
        return info;
      }
    }));
  }, [query]);

  return (
    <div>
      <div className={classes.middlePart}>
        <h2>Harry Potter Characters</h2>
        <p>Source: HP-API!</p>
        <form>
          <input className={classes.search} onChange={updateQuery} ref={inputRef} type="text" placeholder="Search..." name="search" />
        </form>
      </div>
      <div className={classes.itemPart}>
        <div className={classes.card}>
          {result[0] && visible.map((data) => (
            <CharactersHome
              key={data.id}
              id={data.id}
              name={data.names}
              image={data.image}
              actor={data.actorName}
              house={data.house}
              patronus={data.patronus}
              wand={data.wand}
              dateOfBirth={data.dateOfBirth}
              hairColour={data.hairColour}
              eyeColour={data.eyeColour}
            />
          ))}
        </div>
      </div>
    </div>
  );
}

export default Home;
