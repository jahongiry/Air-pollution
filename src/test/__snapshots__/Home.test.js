import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Home from '../../components/Details';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';

it('Builds the snapchot of the Home component Details', () => {
  const three = renderer
    .create(
      <Provider store={store}>
      <Router>
        <Home />
      </Router>
      </Provider>,
    )
    .toJSON();
  expect(three).toMatchSnapshot();
});