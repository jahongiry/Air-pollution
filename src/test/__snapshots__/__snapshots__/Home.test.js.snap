// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Builds the snapchot of the Home component Details 1`] = `
<div
  className="middlePart"
>
  <div>
    <img
      alt="Character"
    />
    <h3 />
    <p>
      House:
      <span>
         
      </span>
    </p>
    <hr />
    <h3 />
  </div>
</div>
`;
